parity-crypto/src/publickey/secret_key.rs

```diff
@@ -52,21 +52,41 @@ impl fmt::Display for Secret {
 
 impl Secret {
 	/// Creates a `Secret` from the given slice, returning `None` if the slice length != 32.
+	/// Caller is responsible to zeroize input slice.
 	pub fn copy_from_slice(key: &[u8]) -> Option<Self> {
 		if key.len() != 32 {
 			return None;
 		}
 		let mut h = H256::zero();
 		h.as_bytes_mut().copy_from_slice(&key[0..32]);
-		Some(Secret { inner: h })
+		Some(Secret { inner: Box::new(h) })
+	}
+
+	/// Creates a `Secret` from the given `str` representation,
+	/// returning an error for hex big endian representation of
+	/// the secret.
+	/// Caller is responsible to zeroize input slice.
+	pub fn copy_from_str(s: &str) -> Result<Self, Error> {
+		let h = H256::from_str(s).map_err(|e| Error::Custom(format!("{:?}", e)))?;
+		Ok(Secret { inner: Box::new(h) })
+	}
+
+	/// Creates a `Secret` from the given inner implementation of
+	/// secret key.
+	/// Caller is responsible to zeroize input secret.
+	pub fn copy_from_inner(key: &key::SecretKey) -> Self {
+		let mut h = H256::zero();
+		h.as_bytes_mut().copy_from_slice(&key[0..SECP256K1_SECRET_KEY_SIZE]);
+		Secret { inner: Box::new(h) }
```

Comment:
```suggestion
```
`copy_from_inner` is now unused?

Reply:
Indeed, I remove all the use of the function and forgot the function itself.